from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(1084, 734)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.document_text_box = QtWidgets.QTextEdit(self.centralwidget)
        self.document_text_box.setGeometry(QtCore.QRect(20, 70, 631, 601))
        self.document_text_box.setObjectName("document_text_box")
        self.numbers_list_view = QtWidgets.QListView(self.centralwidget)
        self.numbers_list_view.setGeometry(QtCore.QRect(680, 70, 381, 101))
        self.numbers_list_view.setObjectName("numbers_list_view")
        self.phone_numbers_btn = QtWidgets.QPushButton(self.centralwidget)
        self.phone_numbers_btn.setGeometry(QtCore.QRect(920, 20, 141, 31))
        self.phone_numbers_btn.setObjectName("phone_numbers_btn")
        self.phone_numbers_label = QtWidgets.QLabel(self.centralwidget)
        self.phone_numbers_label.setGeometry(QtCore.QRect(680, 20, 191, 31))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.phone_numbers_label.setFont(font)
        self.phone_numbers_label.setObjectName("phone_numbers_label")
        self.email_label = QtWidgets.QLabel(self.centralwidget)
        self.email_label.setGeometry(QtCore.QRect(680, 200, 191, 31))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.email_label.setFont(font)
        self.email_label.setObjectName("email_label")
        self.email_btn = QtWidgets.QPushButton(self.centralwidget)
        self.email_btn.setGeometry(QtCore.QRect(920, 200, 141, 31))
        self.email_btn.setObjectName("email_btn")
        self.email_list_view = QtWidgets.QListView(self.centralwidget)
        self.email_list_view.setGeometry(QtCore.QRect(680, 250, 381, 101))
        self.email_list_view.setObjectName("email_list_view")
        self.topic_label = QtWidgets.QLabel(self.centralwidget)
        self.topic_label.setGeometry(QtCore.QRect(680, 380, 191, 31))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.topic_label.setFont(font)
        self.topic_label.setObjectName("topic_label")
        self.topic_btn = QtWidgets.QPushButton(self.centralwidget)
        self.topic_btn.setGeometry(QtCore.QRect(920, 380, 141, 31))
        self.topic_btn.setObjectName("topic_btn")
        self.topis_list_view = QtWidgets.QListView(self.centralwidget)
        self.topis_list_view.setGeometry(QtCore.QRect(680, 430, 381, 101))
        self.topis_list_view.setObjectName("topis_list_view")
        self.document_label = QtWidgets.QLabel(self.centralwidget)
        self.document_label.setGeometry(QtCore.QRect(20, 20, 191, 31))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.document_label.setFont(font)
        self.document_label.setObjectName("document_label")
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1084, 21))
        self.menubar.setObjectName("menubar")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.phone_numbers_btn.setText(_translate("MainWindow", "Run extraction"))
        self.phone_numbers_label.setText(_translate("MainWindow", "Extracted phone numbers"))
        self.email_label.setText(_translate("MainWindow", "Extracted emails"))
        self.email_btn.setText(_translate("MainWindow", "Run extraction"))
        self.topic_label.setText(_translate("MainWindow", "Topic of document"))
        self.topic_btn.setText(_translate("MainWindow", "Find topic"))
        self.document_label.setText(_translate("MainWindow", "Document"))

    def get_document(self):
        return self.document_text_box
